contrib/snap/snapcraft.yaml

```diff
@@ -0,0 +1,236 @@
+      - -include/
+      - -lib/pkgconfig
+      - -lib/python3.5
+      - -sbin/
+      - -share/
+      - -etc/
+      - -lib/*.a
+  libfwup-dev:
+    plugin: make
+    source: https://github.com/rhboot/fwupdate.git
+    source-type: git
+    make-parameters:
+      - EFIDIR=ubuntu
+      - GNUEFIDIR=/usr/lib
+      - LIBDIR=$SNAPCRAFT_STAGE/usr/lib/x86_64-linux-gnu
+      - libdir=/usr/lib/x86_64-linux-gnu
+      - PKG_CONFIG_PATH=$SNAPCRAFT_STAGE/usr/lib/x86_64-linux-gnu/pkgconfig
+      - --eval=export PKG_CONFIG_PATH
+    build-packages:
+      - elfutils
+      - gnu-efi
+      - libasm1
+      - libdw1
+    prime:
+      - -usr/lib/debug
+      - -usr/lib/systemd
+      - -usr/src
+      - -usr/share
+      - -usr/lib/*/pkgconfig
+      - -usr/include
+      - -usr/bin
+    after: [libsmbios, libefivar-fixpkgconfig]
+  meson:
+    plugin: python
+    source: https://github.com/mesonbuild/meson.git
+    source-tag: 0.46.1
+    build-packages:
+      - ninja-build
+    prime:
+      - -bin
+      - -etc
+      - -lib
+      - -share
+      - -usr
+  appstream-glib-dev:
+    plugin: meson
+    meson-parameters: [--prefix=/usr, -Dgtk-doc=false, -Dintrospection=false, -Dman=false, -Drpm=false]
+    source: https://github.com/hughsie/appstream-glib
+    source-type: git
+    build-packages:
+      - python3-pip
+      - gperf
+      - intltool
+      - libarchive-dev
+      - libgcab-dev
+      - libgdk-pixbuf2.0-dev
+      - libgirepository1.0-dev
+      - libglib2.0-dev
+      - libgtk-3-dev
+      - libjson-glib-dev
+      - libsoup2.4-dev
+      - libsqlite3-dev
+      - libyaml-dev
+      - libstemmer-dev
+      - uuid-dev
+      - xmlto
+    stage-packages:
+      - libarchive13
+      - libgcab-1.0-0
+      - libsoup2.4-1
+      - libstemmer0d
+      - libgdk-pixbuf2.0-0
+    prime:
+      - -usr/bin
+      - -usr/include
+      - -usr/share/doc
+      - -usr/lib/*/asb-plugins-5
+      - -usr/share/bash-completion
+      - -usr/share/aclocal
+      - -usr/lib/*/pkgconfig
+      - -usr/share/installed-tests
+      - -usr/lib/systemd
+      - -usr/lib/glib-networking
+      - -usr/lib/dconf
+      - -usr/share/X11
+      - -usr/share/GConf
+      - -usr/share/dbus-1
+      - -usr/share/glib-2.0/schemas
+      - -usr/share/lintian
+      - -usr/share/man
+    after: [meson]
+  fwupd:
+    plugin: meson
+    meson-parameters: [--prefix=/usr,
+                       --libexecdir=/usr/lib,
+                       -Dtests=false,
+                       -Ddaemon=false,
+                       -Dgtkdoc=false,
+                       -Dplugin_colorhug=false,
+                       -Dplugin_uefi_labels=false,
+                       -Dintrospection=false,
+                       -Dsystemd=false,
+                       -Dman=false,
+                       -Dconsolekit=false,
+                       -Dpkcs7=false,
+                       -Dgpg=false]
+    source: https://github.com/hughsie/fwupd
+    source-branch: wip/hughsie/snap-crackle-pop
```

Comment:
You want master, right?

Reply:
Actually, you want $TRAVIS_BRANCH -- which the flatpak manifest also needs to as well. I assume the only way to do this is a manifest.in -> manifest? Ideas welcome.

Reply:
Adjusted for snap.